config/nrfconnect/chip-lib.cmake

```diff
@@ -82,7 +82,7 @@ function(chip_configure TARGET_NAME)
 
     zephyr_get_compile_flags(CHIP_CXXFLAGS CXX)
     list(FILTER CHIP_CXXFLAGS EXCLUDE REGEX -std.*)
-    list(APPEND CHIP_CXXFLAGS "-std=gnu++11")
+    list(APPEND CHIP_CXXFLAGS "-std=gnu++17")
```

Comment:
I would recommend adding a function like:
```
function(chip_select_cpp_standard VAR)
    if (CONFIG_STD_CPP11)
        set(${VAR} -std=gnu++11 PARENT_SCOPE)
    elseif (CONFIG_STD_CPP14)
        set(${VAR} -std=gnu++14 PARENT_SCOPE)
    ... up to c++20
    else()
        message(FATAL_ERROR "Building with unsupported C++ standard")
    endif()
endfunction()
```
and then use it here like:
```
chip_select_cpp_standard(CHIP_CXX_STANDARD)
list(APPEND CHIP_CXXFLAGS ${CHIP_CXX_STANDARD})
```

So that we don't change the standard for apps other than pigweed-app.

Reply:
Cool idea, done